rpc/block_svc_routines.c

```diff
@@ -4375,6 +4394,22 @@ block_create_common(blockCreate *blk, char *control, char *volServer, char *prio
   }
   reply->exit = -1;
 
+  if (GB_ALLOC_N(reply->out, 8192) < 0) {
+    goto out;
+  }
+
+  if (GB_ASPRINTF(&exec, GB_SAVECFG_NAME_CHECK, blk->block_name) == -1) {
+    goto out;
+  }
+
+  save = gbRunnerGetOutput(exec);
+  if (save && atol(save)) {
+    snprintf(reply->out, 8192, GB_IS_ALREADY_EXIST, blk->block_name);
```

Comment:
```
diff --git a/rpc/block_svc_routines.c b/rpc/block_svc_routines.c
index 1fb8d92..2107b69 100644
--- a/rpc/block_svc_routines.c
+++ b/rpc/block_svc_routines.c
@@ -58,6 +58,7 @@
 # define   GB_NODE_IN_USE       224
 # define   GB_BLOCK_NOT_LOADED  225
 # define   GB_BLOCK_NOT_FOUND   226
+# define   GB_BLOCK_NAME_EXIST  227
 
 pthread_mutex_t lock = PTHREAD_MUTEX_INITIALIZER;

```

set reply->exit = GB_BLOCK_NAME_EXIST;

Then defend for this in the daemon's cli thread/process space.

Thanks!

Reply:
Yeah, will fix it.
Thanks.